API.md

```diff
@@ -1,86 +1,57 @@
 # Basic API Guide
 
-## Login
-
-* Endpoint: `/login`
-* Verb: `POST`
-* Requires existing session: `false`
-* POST params:
-    - `email`: string
-    - `password`: string
-* Returns: 200 JSON string (JWT token)
-```json
-"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoiNWZjNTFkOWNhMGYzNTFmYjYyMjkwMDlhIiwidGllciI6MSwiZXhwIjoxNjA2ODQwNjAxLCJpYXQiOjE2MDY3NTQyMDEsImlzcyI6IlNreW5ldCBBY2NvdW50cyJ9.0dhkOMHFBcoZSSBDrVBca5SOSeU3zMEOLVfNQWf_cOI"
-```
-also 400, 401, 422, 500
-
 ## User
 
-### Create
-
-* Endpoint: `/user`
-* Verb: `POST`
-* Requires existing session: `false`
-* POST params:
-    - `email`: string, must be valid, unique to the DB
-    - `password`: string 
-    - `firstName`: string, optional
-    - `lastName`: string, optional
-* Returns: 201 JSON object
-```json
-{
-    "firstName": "John",
-    "lastName": "Doe",
-    "email": "john@doe.com",
-    "tier": 0
-}
-```
-also 400 and 500
+### Get
 
-### Lookup own user data
+This request combines the "get user data" and "create user" requests - if the
```

Comment:
This is an extremely weird API design. `GET` requests don't go and create new entities in the database. 

I'm also confused as to how this would work, don't we need some user information to create a `user` object with?  Like, an email at least? 

How is this request authenticated, it says that it requires a session, how can one acquire a session if he's not a user yet.

Reply:
I'll start in LIFO order because it makes more sense:

> How is this request authenticated, it says that it requires a session, how can one acquire a session if he's not a user yet.

This is something @m-cat also noted, I don't really explain how the authentication works. This is because the document was initially meant entirely for Karol who knows this quite well. I have now added a section to `API.md` that clears it out. For the sake of a complete conversation, I'll repeat some of it here:
 - All requests are authenticated by a JWT token passed in the headers.
 - This token is issued by ORY (I use that in place of switching between Kratos and Oathkeeper).
 - This token holds some infor about the user, most important of which is the `sub` or the user identity in ORY.

> I'm also confused as to how this would work, don't we need some user information to create a user object with? Like, an email at least?

No, all such information is stored in ORY. The only thing we need in our DB is the `sub` of the user or their unique id from ORY. That is enough for us to uniquely identify them without needlessly duplicating the user data already stored in ORY (email, names, etc.).

> This is an extremely weird API design. GET requests don't go and create new entities in the database.

We are stepping away from the normal RESTful path here. The reason for that is that we'll probably never work with entity ids and all that - we'll always operate on the current user which we can get from the JWT in the request. 

We can also split this into three calls (`GET /user` -> 404, `POST /user` without any additional data because we don't need it -> 201, `GET /user` -> 200) but that doesn't seem like a really useful flow to me. REST is a great design pattern but doesn't necessarily work well everywhere. For this particular case folding these three calls into a single one seems to work well. If we want to split them in the future we're very much free to do it.

Reply:
I think a cleaner approach would be to use a hook that inserts the database upon registration. You can resolve the comment, but I do think it's not the cleanest approach. Perhaps we should resolve in a follow-up.

Reply:
F/U: https://github.com/NebulousLabs/skynet-accounts/issues/19